### PR TITLE
Tm2 upgrade db update

### PR DESCRIPTION
**Pull request by OTjornelund**
- Upgraded DbUpdate Portal to .NET 6
- Improved NVD Suppressions
- Updated DbUpdate Portal's pipeline .yml file to use .NET 6

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -331,11 +331,11 @@ ASALocalRun/
 **/Portal/wwwroot/lib/*
 
 # NCNE Portal wwwroot lib
-**/NCNEPortal.OLD/wwwroot/lib/*
 **/NCNEPortal/wwwroot/lib/*
 
 # DbUpdate Portal wwwroot lib
 **/DbUpdatePortal/wwwroot/lib/*
+**/DbUpdatePortal.OLD/wwwroot/lib/*
```
Comment (uhakdt) [inline .gitignore:338]: Maybe remove this now?

Comment (SamirHassoun) [reply .gitignore]: Fixed, thanks

**Review by uhakdt — APPROVED** [review]

File: src/Databases/DbUpdateWorkflowDatabase.Tests/Helpers/AdUserHelper.cs [hunk 2]
```diff
@@ -18,7 +15,7 @@ internal static AdUser CreateTestUser(DbUpdateWorkflowDbContext dbContext, int u
                 name = $"Test User {uniqueNumber}";
             }
 
-            var users = dbContext.AdUser.ToList();
+            var users = dbContext.AdUser.AsNoTracking().ToList();
```
Comment (UKHOGreg) [inline src/Databases/DbUpdateWorkflowDatabase.Tests/Helpers/AdUserHelper.cs:18]: Has the addition of `AsNoTracking()` been checked to make sure there are no later EF update operations for the returned `AdUser` entity?

File: src/Databases/NCNEWorkflowDatabase.Tests/Helpers/AdUserHelper.cs [hunk 3]
```diff
@@ -14,7 +16,7 @@ internal static AdUser CreateTestUser(NcneWorkflowDbContext dbContext, int uniqu
                 name = $"Test User {uniqueNumber}";
             }
 
-            var users = dbContext.AdUser.ToList();
+            var users = dbContext.AdUser.AsNoTracking().ToList();
```
Comment (UKHOGreg) [inline src/Databases/NCNEWorkflowDatabase.Tests/Helpers/AdUserHelper.cs:19]: Has the addition of `AsNoTracking()` been checked to make sure there are no later EF update operations for the returned `AdUser` entity?

File: src/DbUpdatePortal/Startup.cs [hunk 4]
```diff
@@ -120,7 +107,9 @@ public void ConfigureServices(IServiceCollection services)
             services.AddScoped<ICarisProjectHelper, CarisProjectHelper>();
             services.AddScoped<IStageTypeFactory, StageTypeFactory>();
             services.AddScoped<IPageValidationHelper, PageValidationHelper>();
-            services.AddScoped<ICommentsHelper, CommentsHelper>();
+            services.AddScoped<ICommentsHelper, CommentsHelper>(); 
+            services.AddAzureAppConfiguration();
```
Comment (UKHOGreg) [inline src/DbUpdatePortal/Startup.cs:111]: Just wondering why this line was introduced?

Is it because of .NET 6?

**Review by UKHOGreg — APPROVED** [review]

File: src/DbUpdatePortal/Startup.cs (hunk 4 above)
Comment (SamirHassoun) [reply src/DbUpdatePortal/Startup.cs:111]: AddAzureAppConfiguration seems to be required for .Net 6

File: src/Databases/DbUpdateWorkflowDatabase.Tests/Helpers/AdUserHelper.cs (hunk 2 above)
Comment (SamirHassoun) [reply src/Databases/DbUpdateWorkflowDatabase.Tests/Helpers/AdUserHelper.cs:18]: AsNoTracking, seems to be needed for unit tests, we did the same thing for NCNE; there is a good chance DB-Assessment unit-tests will need it too

**Pull request merged**